kaldb/src/main/java/com/slack/kaldb/elasticsearchApi/ZipkinService.java

```diff
@@ -0,0 +1,186 @@
+package com.slack.kaldb.elasticsearchApi;
+
+import static com.slack.kaldb.util.JsonUtil.writeJsonValues;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.linecorp.armeria.server.annotation.*;
+import com.slack.kaldb.elasticsearchApi.searchResponse.*;
+import com.slack.kaldb.proto.service.KaldbServiceGrpc;
+import com.slack.kaldb.util.JsonUtil;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Elasticsearch compatible API service, for use in Grafana
+ *
+ * @see <a href="https://zipkin.io/zipkin-api/#/">Zipkin API</a>
+ */
+@SuppressWarnings(
+    "OptionalUsedAsFieldOrParameterType") // Per https://armeria.dev/docs/server-annotated-service/
+public class ZipkinService {
+  private final KaldbServiceGrpc.KaldbServiceImplBase searcher;
+
+  public ZipkinService(KaldbServiceGrpc.KaldbServiceImplBase searcher) {
+    this.searcher = searcher;
+  }
+
+  @JsonFormat(shape = JsonFormat.Shape.ARRAY)
+  public static final class GetServicesResult {
+
+    private final List<String> services;
+
+    public GetServicesResult(List<String> services) {
+      this.services = services;
+    }
+
+    @JsonProperty("services")
+    public List<String> services() {
+      return services;
+    }
+  }
+
+  public static final class Service {
```

Comment:
same with this and the span class - we can move this out to their individual files.